server/config/router.js

```diff
@@ -58,33 +58,56 @@ module.exports = function (nextjs, koa, config) {
     Redirects
   \******************************************************************************/
 
-  // Legacy blog list route
-  router.get('/blogs', async (ctx, next) => {
+  // Legacy Wordpress permalinks
+  // e.g. /2017/09/07/universal-service-a-fuel-rats-thargoid-cartoon
+  router.get('/:year/:month/:day/:slug', async (ctx, next) => {
+    let {
+      day,
+      month,
+      slug,
+      year,
+    } = ctx.params
+
+    let dayIsValid = parseInt(day) && (day.length === 2)
+    let monthIsValid = parseInt(month) && (month.length === 2)
+    let yearIsValid = parseInt(year) && (year.length === 4)
+
+    if (dayIsValid && monthIsValid && yearIsValid) {
+      ctx.status = 301
+      return await ctx.redirect(`/blog/${slug}`)
+    }
+
+    await next()
+  })
+
+  // Permanent Redirects
+  router.all('/blogs', async (ctx, next) => {
     ctx.status = 301
     await ctx.redirect(`/blog`)
   })
 
-  router.get('/fuel-rats-lexicon', async (ctx, next) => {
+  router.all('/get-help', async (ctx, next) => {
     ctx.status = 301
-    await ctx.redirect(`https://confluence.fuelrats.com/pages/viewpage.action?pageId=3637257`)
+    await ctx.redirect(`/i-need-fuel`)
   })
 
-  router.get('/get-help', async (ctx, next) => {
+  router.all('/fuel-rats-lexicon', async (ctx, next) => {
     ctx.status = 301
-    await ctx.redirect(`/i-need-fuel`)
+    await ctx.redirect(`https://confluence.fuelrats.com/pages/viewpage.action?pageId=3637257`)
   })
 
-  router.get('/privacy-policy', async (ctx, next) => {
+  // Temporary Redirects
+  router.all('/privacy-policy', async (ctx, next) => {
     ctx.status = 307
     await ctx.redirect(`https://confluence.fuelrats.com/display/FRKB/Privacy+Policy`)
   })
 
-  router.get('/terms-of-service', async (ctx, next) => {
+  router.all('/terms-of-service', async (ctx, next) => {
     ctx.status = 307
     await ctx.redirect(`https://confluence.fuelrats.com/display/FRKB/Terms+of+Service`)
   })
 
-  router.get('/code-of-conduct', async (ctx, next) => {
+  router.all('/code-of-conduct', async (ctx, next) => {
```

Comment:
Why did we change these to `router.all` instead of `router.get`?

Reply:
Being that these routes are meant to be redirected, capturing all methods ensures that there isn't any unexpected behavior (in case someone decides to snoop around :P). For these specific routes it really doesn't matter, but it does improve consistency between 301s on nginx and 301s set by the server router. It's more a thinking ahead preventative measure than anything else.

Reply:
Fair enough.